testnet/stacks-node/src/mockamoto.rs

```diff
@@ -824,6 +824,9 @@ impl MockamotoNode {
             Some(AddressHashMode::SerializeP2PKH),
         );
 
+        let public_key = StacksPublicKey::from_private(&self.miner_key);
```

Comment:
I think there instead needs to be a separate `signing_key` field in `MockamotoNode`.  Same for the Nakamoto node.  Perhaps it can be part of the `Keychain`?